app.rb

```diff
@@ -0,0 +1,91 @@
+require 'sinatra/base'
+require 'sinatra/reloader'
+require './lib/database_connection'
+require './lib/user_repository'
+require './lib/user'
+require './lib/peep_repository'
+require './lib/peep'
+
+DatabaseConnection.connect('chitter_test')
+
+class Application < Sinatra::Base
+  # This allows the app code to refresh
+  # without having to restart the server.
+  configure :development do
+    register Sinatra::Reloader
+  end
+
+  enable :sessions
+  
+  get '/' do
+    repo = PeepRepository.new
+    @peeps = repo.all.sort_by(&:post_time).reverse
+
+    return erb :index, :layout => :main_layout if session[:user_id].nil?
+    return erb :index, :layout => :main_layout
```

Comment:
Is this repeated so you can insert a custom page for if a user is not logged in?
Otherwise it looks like you only need line 25 here.